cpp_utils/src/cpp/math/random/RandomManager.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file math_extension.cpp
+ *
+ */
+
+#include <cstdlib>
+
+#include <cpp_utils/math/random/RandomManager.hpp>
+
+namespace eprosima {
+namespace utils {
+
+RandomManager::RandomManager(const RandomNumberType original_seed /* = 1 */ )
+    : std_random_generator_(original_seed)
+{
+    // Do nothing
+}
+
+template <>
+CPP_UTILS_DllAPI RandomNumberType RandomManager::rand<true> () noexcept
+{
+    return pure_rand();
+}
+
+template <>
+CPP_UTILS_DllAPI RandomNumberType RandomManager::rand<false> () noexcept
+{
+    return sequence_rand();
+}
+
+RandomNumberType RandomManager::pure_rand () noexcept
+{
+    return pure_random_generator_();
+}
+
+RandomNumberType RandomManager::sequence_rand () noexcept
+{
+    return std_random_generator_();
+}
+
+RandomNumberType RandomManager::rand (const RandomNumberType seed) noexcept
```

Comment:
First, it is intended to be a Singleton. And secondly, I do not understand the doubt :thinking:

Reply:
I just cannot imagine a practical use for this method, generating only one number given a specific seed. Perhaps being able to reset the seed might be useful.

Reply:
using a seed to generate a random number is a very common (if not the most common) use of rand. It would be very useful for testing, where pseudorandom values are desired, but keeping them always the same